cpp/include/raft/random/detail/rng_impl.cuh

```diff
@@ -451,8 +480,33 @@ struct PhiloxGenerator {
     return ret;
   }
 
-  DI void next(float& ret) { ret = curand_uniform(&(this->philox_state)); }
-  DI void next(double& ret) { ret = curand_uniform_double(&(this->philox_state)); }
+  DI float next_float()
+  {
+    float ret;
+    uint32_t val = next_u32() >> 8;
+    ret          = static_cast<float>(val) / (1U << 24);
```

Comment:
Minor nitpick: there is an implicit conversion from `unsigned int` to `float` here, which I would make explicit: `static_cast<float>(1U << 24)` or `float(1U << 24)`.
This is similar in the double case. Additionally, in double I would use `long long unsigned` since `long` is not guaranteed to have 64 bits in all C++ implementations: `static_cast<double>(1LLU << 53)`

Reply:
Good catch, I'll make these changes. Let's use `uint64_t` to be absolutely sure.